iamclient/iamclient.go

```diff
@@ -256,23 +252,68 @@ func (client *Client) Close() (err error) {
  * Private
  **********************************************************************************************************************/
 
-func (client *Client) getNodeInfo() (nodeID, nodeType string, err error) {
+func (client *Client) getNodeInfo() error {
 	ctx, cancel := context.WithTimeout(context.Background(), iamRequestTimeout)
 	defer cancel()
 
 	response, err := client.publicService.GetNodeInfo(ctx, &empty.Empty{})
 	if err != nil {
-		return "", "", aoserrors.Wrap(err)
+		return aoserrors.Wrap(err)
 	}
 
 	log.WithFields(log.Fields{
 		"nodeID":   response.GetNodeId(),
 		"nodeType": response.GetNodeType(),
 	}).Debug("Get node Info")
 
-	return response.GetNodeId(), response.GetNodeType(), nil
-}
+	client.nodeInfo = cloudprotocol.NodeInfo{
```

Comment:
there were some convertion functions in aos_common, can be utilized here I think.